File: src/com/derry/s6/KtBase128.java
package com.derry.s6;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class KtBase128 {

    public static void main(String[] args) {
        // 1.定义name集合
        List<String> names = new ArrayList<>();
        names.add("Zhangsan");
        names.add("Lisi");
        names.add("Wangwu");

        // 2.定义age集合
        List<Integer> ages = new ArrayList<>();
        ages.add(20);
        ages.add(21);
        ages.add(22);

        // 3.合并以上两个集合
        Map<String, Integer> newMap = new HashMap<>();
        for (int i = 0; i < names.size(); i++) {
            newMap.put(names.get(i), ages.get(i));
        }

        // 4.给集合添加详细内容，方便输出
        List<String> showList = new ArrayList<>();
        for (Map.Entry<String, Integer> stringIntegerEntry : newMap.entrySet()) {
            String result = String.format("you name:%s, you age:%d",
                                           stringIntegerEntry.getKey(), stringIntegerEntry.getValue());
            showList.add(result);
        }

        // 5.输出最后的成果 结果
        for (int i = 0; i < showList.size(); i++) {
            System.out.println(showList.get(i));
        }
    }

}

File: src/com/derry/s6/KtBase128.kt
package com.derry.s6

// TODO 128-Kotlin语言中使用函数式编程
/*
fun main() {
    // 1.定义name集合
    val names = listOf("Zhangsan", "Lisi", "Wangwu")
    // 2.定义age集合
    val ages = listOf(20, 21, 22)
    // 3.合并以上两个集合
    // 4.给集合添加详细内容，方便输出
    // 5.输出最后的成果 结果
    names.zip(ages).toMap().map { "you name:${it.key}, you age:${it.value}" }.map { println(it) }
}
*/

fun main() {
    listOf("Zhangsan", "Lisi", "Wangwu").zip(listOf(20, 21, 22)).toMap().map { println("you name:${it.key}, you age:${it.value}") }
}


